### PR TITLE
AzureMonitor: select plugin route from cloudname

### LINKED ISSUE #27243: Can not get resource metics data from Azure China
**What happened**: All resource metric data are NULL.

**What you expected to happen**: show the graph of resource metric

**How to reproduce it (as minimally and precisely as possible)**: configurate azure monitor for Azure china 

**Anything else we need to know?**:

**Environment**:
- Grafana version: 7.0+
- Data source type & version: Azure monitor for Azure China
- OS Grafana is installed on: CentOS 7
- User OS & Browser: MACos & Chrome
- Grafana plugins: N/A
- Others:
image

### PR DESCRIPTION
**Pull request by kylebrandt**
**What this PR does / why we need it**:
For the Azure Monitor service the Azure Monitor data source, selects the plugin route.

**Which issue(s) this PR fixes**:
maybe fixes #27243

**Notes for Reviewer**
https://github.com/grafana/grafana/issues/27243#issuecomment-682910259 explains this

### CODE REVIEW

File: pkg/tsdb/azuremonitor/azuremonitor-datasource.go [hunk 1]
```diff
@@ -223,15 +223,15 @@ func (e *AzureMonitorDatasource) createRequest(ctx context.Context, dsInfo *mode
 		return nil, errors.New("Unable to find datasource plugin Azure Monitor")
 	}
 
+	cloudName := dsInfo.JsonData.Get("cloudName").MustString("azuremonitor")
 	var azureMonitorRoute *plugins.AppPluginRoute
 	for _, route := range plugin.Routes {
-		if route.Path == "azuremonitor" {
+		if strings.Contains(route.Path, cloudName) {
```
Comment (marefr) [inline pkg/tsdb/azuremonitor/azuremonitor-datasource.go:229]: Shouldn't it still be equals here?

**Review by marefr — COMMENTED** [review]
Left a comment

File: pkg/tsdb/azuremonitor/azuremonitor-datasource.go (hunk 1 above)
Comment (kylebrandt) [reply pkg/tsdb/azuremonitor/azuremonitor-datasource.go]: Yes, I think somehow I confused myself when looking at the plugin json and where cloudName comes from in the context of the different services.

**Review by marefr — APPROVED** [review]
LGTM

Given the referenced issue is making a request to the backend and not going through the data proxy this should make it work better :+1:

**Pull request merged**